blog-posts/step-function-learnings/step-function-logs-learning.md

```diff
@@ -0,0 +1,88 @@
+
+All you need is to create a log group, configure it and then attach it to the State Machine via the logs prop. If you want to jump directly to the snippet [here](#how-to-configure-logs-in-an-express-step-function-in-cdk) is a shortcut.
+
+The cdk handles all the work of creating the role for the step function to write inside CloudWatch.
+
+The LogLevel allows you to select what kind of information you want to have. The four levels are: `OFF`, `ALL`, `ERROR` and `FATAL`.
+
+But the `includeExecutionData` property of the CDK construct makes all the difference. Thanks to this property set to `true`, you have all step transition data displayed in the Step Function console.
+
+|                                                                                               Express with NO *logs execution data*                                                                                                |                      Express with *logs execution data* because `includeExecutionData` is set to `true`                       |
+| :--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: | :---------------------------------------------------------------------------------------------------------------------------: |
+| ![Express with logs execution data NOT logged because no includeExecutionData](./assets/express-with-logs-but-not-includeExecutionData-exe-logs.png 'Express with logs execution data NOT logged because no includeExecutionData') | ![Express with logs execution data logged](./assets/express-with-logs-exe-logs.png 'Express with logs execution data logged') |
+
+### How to configure logs in an Express Step Function in CDK
+
+This is pretty easy and straightforward but you just need to know it.
+
+```ts
+import { App, RemovalPolicy, Stack } from 'aws-cdk-lib';
+import { LogGroup, RetentionDays } from 'aws-cdk-lib/aws-logs';
+import { LogLevel, Pass, StateMachine, StateMachineType } from 'aws-cdk-lib/aws-stepfunctions';
+
+export class ArticleStack extends Stack {
+  constructor(scope: App, id: string) {
+    super(scope, id);
+
+    const expressLogGroup = new LogGroup(this, 'ExpressLogs', {
+      retention: RetentionDays.ONE_DAY,
+      removalPolicy: RemovalPolicy.DESTROY,
+    });
+
+    new StateMachine(this, 'ExpressWithLogs', {
+      definition: new Pass(scope, 'ExampleStepForExpressWithLogs', {
+        parameters: {
+          bar: 'foo',
+        },
+      }),
+      stateMachineType: StateMachineType.EXPRESS,
+      logs: {
+        destination: expressLogGroup,
+        level: LogLevel.ALL,
+        includeExecutionData: true,
+      },
+    });
+  }
+}
+```
+
+For this snippet of code, I used the V2 of the CDK (2.56 to be more precise)
+
+## Conclusion
+
+You are all set to have logs on all your Step Functions in CDK.
```

Comment:
AWS Step Functions c'est le service, je remplacerais par "on all you state machines"